lxd/device/nic_bridged.go

```diff
@@ -1372,7 +1380,15 @@ func (d *nicBridged) setupOVSBridgePortVLANs(hostName string) error {
 
 	// Add any tagged VLAN memberships.
 	if d.config["vlan.tagged"] != "" {
-		vlanIDs := util.SplitNTrimSpace(d.config["vlan.tagged"], ",", -1, true)
+		intNetworkVLANs, err := ExpandNetworkVLANList(util.SplitNTrimSpace(d.config["vlan.tagged"], ",", -1, true))
+		var vlanIDs []string
+
+		for _, intNetworkVLAN := range intNetworkVLANs {
+			vlanIDs = append(vlanIDs, strconv.Itoa(intNetworkVLAN))
+		}
+		if err != nil {
```

Comment:
This error check should be immediately after the `ExpandNetworkVLANList` call, not here.